client/src/components/RecipeShow.js

```diff
@@ -11,26 +16,140 @@ const RecipeShow = (props) => {
+          if(ingredientInPantry.unit === ingredient.unit) {
+            const updatedIngredient = {
+              ...ingredientInPantry,
+              amount: ingredientInPantry.amount - ingredient.amount
+            }
+            return updatedIngredient
+            } else {
+            const convertedRecipeIngredient = Ingredients.convertIngredient(ingredient)
+            const convertedPantryIngredient = Ingredients.convertIngredient(ingredientInPantry)
+            if (!convertedRecipeIngredient || !convertedPantryIngredient) {
+              return false
+            } else {
+              const newTotal = Ingredients.convertAndUpdate(ingredientInPantry, ingredient)
+              const updatedIngredient = {
+                ...ingredientInPantry,
+                amount: newTotal
+              }
+              return updatedIngredient
+            }
+          }
+        } else {
+          return false
+        }
+      })
+      setUpdatedIngredients(ingredientsToUpdate)
+    }
+    
+    const compareIngredient = (ingredient) => {
+      const ingredientInPantry = pantryIngredients.find(pantryIngredient => pantryIngredient.name.toLowerCase() === ingredient.name)
+      if(ingredientInPantry) {
+        if(ingredientInPantry.unit === ingredient.unit) {
+          if(ingredientInPantry.amount >= ingredient.amount) {
+            return "have enough"
+          } else {
+            return "need more"
+          }
+        } else {
+          const convertedRecipeIngredient = Ingredients.convertIngredient(ingredient)
+          const convertedPantryIngredient = Ingredients.convertIngredient(ingredientInPantry)
+          if (!convertedRecipeIngredient || !convertedPantryIngredient) {
+            return "can't tell"
+          } else {
+            if (convertedPantryIngredient - 3 >= convertedRecipeIngredient) {
+              return "have enough"
+            } else if (convertedPantryIngredient + 3 <= convertedRecipeIngredient) {
+              return "need more"
+            } else {
+              return "it's close"
+            }
+          }
+        }
+      } else {
+        return "don't have"
+      }
+    }
+
+    const makeRecipe = async (event) => {
+      let updatedIngredientData = updatedIngredients.filter(element => element)
+      updatedIngredientData.forEach(ingredient => {
+        ingredient.amount = Number(ingredient.amount.toFixed(2))
+        if(ingredient.amount <= 0) {
+          ingredient.amount = 0.01
+        }
+      })
+      try {
+        const response = await fetch("/api/v1/recipes/make", {
+          method: "POST",
+          headers: new Headers({
+            "Content-Type": "application/json"
+          }),
+          body: JSON.stringify(updatedIngredientData)
+        })
+        if(response.ok) {
+          fetchPantryIngredients()
+          setMade(true)
+        }
+      } catch(err) {
+        console.error(`Error in Fetch: ${err.message}`)
+      }
+    }
```

Comment:
wonderful work for this! i nthe future, it would be great to also ask the user about how the recipe turned out!